ts/session/apis/snode_api/swarmPolling.ts

```diff
@@ -702,14 +881,214 @@ export class SwarmPolling {
         '[pollOnceForOurDisplayName] after merging there are no user config messages'
       );
     }
-    const displayName = await this.handleSharedConfigMessages(userConfigMessagesMerged, true);
+    let displayNameFound: string | undefined;
+    try {
+      const keypair = await UserUtils.getUserED25519KeyPairBytes();
+      if (!keypair || !keypair.privKeyBytes) {
+        throw new Error('edkeypair not found for current user');
+      }
+
+      const privateKeyEd25519 = keypair.privKeyBytes;
+
+      // we take the latest config message to create the wrapper in memory
+      const incomingConfigMessages = userConfigMessagesMerged.map(m => ({
+        data: StringUtils.fromBase64ToArray(m.data),
+        hash: m.hash,
+      }));
+
+      await UserConfigWrapperActions.init(privateKeyEd25519, null);
+      await UserConfigWrapperActions.merge(incomingConfigMessages);
+
+      const foundName = await UserConfigWrapperActions.getName();
+      if (!foundName) {
+        throw new Error('UserInfo not found or name is empty');
+      }
+      displayNameFound = foundName;
+    } catch (e) {
+      window.log.warn('LibSessionUtil.initializeLibSessionUtilWrappers failed with', e.message);
+    } finally {
+      await UserConfigWrapperActions.free();
+    }
 
-    if (isEmpty(displayName)) {
+    if (!displayNameFound || isEmpty(displayNameFound)) {
       throw new NotFoundError(
         '[pollOnceForOurDisplayName] Got a config message from network but without a displayName...'
       );
     }
 
-    return displayName;
+    return displayNameFound;
+  }
+}
+
+// zod schema for retrieve items as returned by the snodes
+const retrieveItemSchema = z.object({
```

Comment:
this will have to be part of a bigger change where we use zod schemas for everything. Currently the type of RetrieveMessageItem and `typeof retrieveItemSchema` do not match, and I do not want to fix it as part of groups.